lib/linkbot/plugin.rb

```diff
@@ -31,33 +31,56 @@ def self.collect(paths)
           end
         end
       end
+      @@plugins = []
+      ObjectSpace.each_object(Class).select { |klass| klass < self }.each do |plugin|
```

Comment:
This scan of `ObjectSpace` to register plugins makes it hard during testing to control the plugin registry the way we expect to be able to for the specs for `Linkbot::Plugin` parent class.

For example, I added a spec for the super important `Bubs` plugin:

```
Bubs
  responds with ⓑⓤⓑⓑⓛⓔ ⓣⓔⓧⓣ

Linkbot::Plugin
  has the correct plugin (FAILED - 1)
  handles a message (FAILED - 2)
...
Failures:

  1) Linkbot::Plugin has the correct plugin
     Failure/Error: expect(Linkbot::Plugin.plugins.length).to eq 1

       expected: 1
            got: 2

       (compared using ==)
     # ./spec/plugins_spec.rb:13:in `block (2 levels) in <top (required)>'

  2) Linkbot::Plugin handles a message
     Failure/Error: puts plugin.messages
     NoMethodError:
       undefined method `messages' for #<Bubs:0x007fe308c7b058>
     # ./spec/plugins_spec.rb:21:in `block (2 levels) in <top (required)>'

```

`Linkbot::Plugin.plugins` looks like ...

```
[ #<Bubs:0x007fe3094b1f10>,
  #<MockPlugin:0x007fe3094b1e70> ]
```

... during the first test. And the `messages` method was added to `MockPlugin` for testing purposes, so `Bubs` doesn't have it for the second failure.

I suppose some choices are

1. figure out some way control what gets included from `ObjectSpace` in the plugin registry during testing which probably involves more logic in the `#collect` method
2. make the spec for `Linkbot::Plugin` that tests generic message handling not care about the number of plugins registered or the number of messages returned from plugins, but uses `include?` to search for an expected message in the message output array. Might lead to unexpected output matches or misbehaving plugins affecting the tests of the parent class